contracts/src/v2/Types.sol

```diff
@@ -0,0 +1,97 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: 2023 Snowfork <hello@snowfork.com>
+pragma solidity 0.8.25;
+
+import {OperatingMode} from "./../types/Common.sol";
+
+// Inbound message from a Polkadot parachain (via BridgeHub)
+struct InboundMessage {
+    // origin
+    bytes32 origin;
+    // Message nonce
+    uint64 nonce;
+    // Commands
+    Command[] commands;
+}
+
+struct Command {
+    uint8 kind;
+    uint64 gas;
+    bytes payload;
+}
+
+library CommandKind {
+    uint8 constant Upgrade = 0;
+    uint8 constant SetOperatingMode = 1;
+    uint8 constant UnlockNativeToken = 2;
+    uint8 constant RegisterForeignToken = 3;
+    uint8 constant MintForeignToken = 4;
+    uint8 constant CreateAgent = 5;
+    uint8 constant CallContract = 6;
+}
```

Comment:
Is this only used for testing?

Reply:
No, these contants are used in the dispatcher in `Gateway.v2_submit`